docs/book/v1/user-repository.md

```diff
@@ -42,3 +42,80 @@ result will be a `UserInterface` instance, otherwise a null value is returned.
 The second function is `getRolesFromUser()` and it specifies how to retrieve
 the roles for a user. If a user does not have roles, this function will return
 an empty array.
+
+
+## Configure the user repository
+
+In order to use a user repository adapter, we need to configure it. For instance,
+to consume a `htpasswd` file, we need to specify the path of the file.
+The configuration is provided in the `['config']['authentication']` value of
+your [PSR-11](http://www.php-fig.org/psr/psr-11/) container.
+
+Using [Expressive](https://docs.zendframework.com/zend-expressive/), this
+configuration can be stored in a file under the `/config/autoload` folder.
+We suggest to use a `.local.php` suffix, like `/config/autoload/auth.local.php`,
+in order to use a local configuration that is not stored in the version control
+system.
+
+You can also provide this configuration using the [ConfigProvider.php](https://github.com/zendframework/zend-expressive-authentication/blob/master/src/ConfigProvider.php)
+class. For more information, you can read this [blog post](https://framework.zend.com/blog/2017-04-20-config-aggregator.html).
+
+
+## htpasswd configuration
+
+The configuration of the `htpasswd` adapter requires only the file path, as
+follows:
+
+```php
+'config' => [
+    'authentication' => [
+        'htpasswd' => 'insert the path to htpasswd file'
```

Comment:
I would add everywhere trailing commas in arrays in the docs as well (this is going to be a part of new coding-standard and it would be good to have it also in configuration, because folks could copy from there)